Added docs for sink topic partitions and replica defaults. (#2547)
confluentinc/ksql: docs/installation/upgrading.rst

Upgrade one KSQL server at a time (i.e. rolling restart). The remaining KSQL servers should have sufficient spare
capacity to take over temporarily for unavailable, restarting servers.

Upgrading from KSQL 5.2 to KSQL 5.3
-----------------------------------

Notable changes in 5.3:

* Configuration:

>     * ksql.sink.partitions and ksql.sink.replicas are deprecated. All new queries will use the source topic partition count and replica count for the sink topic instead unless partitions and replicas are set in the WITH clause.

Reviewer: ```suggestion
    * ``ksql.sink.partitions`` and ``ksql.sink.replicas`` are deprecated. All new queries use the source topic partition count and replica count for the sink topic, unless you set partitions and replicas explicitly in the WITH clause.
```
